### PR TITLE
watermark: avoid mixed precision arithmetic, remove redundant code

### PR DESCRIPTION
**Pull request by rabauke**

### CODE REVIEW

File: src/iop/watermark.c [hunk 1]
```diff
@@ -993,7 +993,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
 
   /* render surface on output */
   guint8 *sd = image;
-  float opacity = data->opacity / 100.0;
+  float opacity = data->opacity / 100.0f;
```
Comment (TurboGit) [inline src/iop/watermark.c]: const

File: src/iop/watermark.c [hunk 2]
```diff
@@ -970,8 +970,8 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
   cairo_translate(cr, tx * roi_out->scale, ty * roi_out->scale);
 
   // compute the center of the svg to rotate from the center
-  float cX = svg_width / 2.0 * roi_out->scale;
-  float cY = svg_height / 2.0 * roi_out->scale;
+  float cX = svg_width / 2.0f * roi_out->scale;
```
Comment (TurboGit) [inline src/iop/watermark.c]: const

File: src/iop/watermark.c [hunk 3]
```diff
@@ -970,8 +970,8 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
...
+  float cY = svg_height / 2.0f * roi_out->scale;
```
Comment (TurboGit) [inline src/iop/watermark.c]: const

File: src/iop/watermark.c [hunk 4]
```diff
@@ -938,10 +938,10 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
 
   // compute bounding box of rotated watermark
   float bb_width, bb_height;
-  bb_width = fabs(svg_width * cos(angle)) + fabs(svg_height * sin(angle));
-  bb_height = fabs(svg_width * sin(angle)) + fabs(svg_height * cos(angle));
-  float bX = bb_width / 2.0 - svg_width / 2.0;
-  float bY = bb_height / 2.0 - svg_height / 2.0;
+  bb_width = fabsf(svg_width * cosf(angle)) + fabsf(svg_height * sinf(angle));
+  bb_height = fabsf(svg_width * sinf(angle)) + fabsf(svg_height * cosf(angle));
+  float bX = bb_width / 2.0f - svg_width / 2.0f;
```
Comment (TurboGit) [inline src/iop/watermark.c]: const

File: src/iop/watermark.c [hunk 5]
```diff
@@ -938,10 +938,10 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
...
+  float bY = bb_height / 2.0f - svg_height / 2.0f;
```
Comment (TurboGit) [inline src/iop/watermark.c]: const

File: src/iop/watermark.c [hunk 6]
```diff
@@ -1002,11 +1002,11 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
   for(int j = 0; j < roi_out->height; j++)
     for(int i = 0; i < roi_out->width; i++)
     {
-      float alpha = (sd[3] / 255.0) * opacity;
+      float alpha = (sd[3] / 255.0f) * opacity;
```
Comment (TurboGit) [inline src/iop/watermark.c]: can be set const

**Comment by rabauke** [comment]
Added `const` qualifiers, applied some further clean-up, fixed a memory leak in `_string_escape`.

**Comment by TurboGit** [comment]
Perfect thanks!

**Pull request merged**